### PR TITLE
Switch all links to HTTPS

### PR DESCRIPTION
**Pull request by aried3r**

### CODE REVIEW

File: BUILD_DETAIL.md [hunk 1]
```diff
@@ -68,8 +68,9 @@ $ bin/cucumber
 
 ## YARD documentation
 
-RSpec uses [YARD](http://yardoc.org/) for API documentation on the [rspec.info site](http://rspec.info/).
-Our commitment to [SemVer](htp://semver.org) requires that we explicitly
+RSpec uses [YARD](https://yardoc.org/) for API documentation on the [rspec.info
+site](https://rspec.info/).
+Our commitment to [SemVer](https://semver.org) requires that we explicitly
```
Comment (aried3r) [inline BUILD_DETAIL.md]: Reformatted so it's 80 char width like the rest of the document.

File: features/README.md [hunk 2]
```diff
@@ -41,9 +41,8 @@ file.
...
-specific features you'd like to see added, find the existing documentation
-incomplete or confusing, or, better yet, wish to write a missing Cucumber
-feature yourself, please [submit an
-issue](http://github.com/rspec/rspec-rails/issues) or a [pull
-request](http://github.com/rspec/rspec-rails).
+Cucumber features over time, and clarifying existing ones.  If you have specific
+features you'd like to see added, find the existing documentation incomplete or
+confusing, or, better yet, wish to write a missing Cucumber feature yourself,
+please [submit an issue](https://github.com/rspec/rspec-rails/issues) or a [pull
+request](https://github.com/rspec/rspec-rails).
```
Comment (aried3r) [inline features/README.md]: Reformatted to 80-char width

File: script/functions.sh [hunk 3]
```diff
@@ -6,7 +6,7 @@ source $SCRIPT_DIR/travis_functions.sh
 source $SCRIPT_DIR/predicate_functions.sh
 
 # If JRUBY_OPTS isn't set, use these.
-# see http://docs.travis-ci.com/user/ci-environment/
+# see https://docs.travis-ci.com/user/reference/overview/
```
Comment (aried3r) [inline script/functions.sh]: This is where the old link redirects right now. It doesn't mention `JRUBY_OPTS` or env variables at all, so I'm wondering if that's the correct link?

File: features/README.md (hunk 2 above)
Comment (JonRowe) [reply features/README.md]: Can you undo this, its formatted for relish.

File: script/functions.sh (hunk 3 above)
Comment (JonRowe) [reply script/functions.sh]: It'll do

File: BUILD_DETAIL.md (hunk 1 above)
Comment (JonRowe) [reply BUILD_DETAIL.md]: Preferred as was.

**Comment by aried3r** [comment]
I have addressed the issues you mentioned. :)

**Pull request merged**